mnixo/kindly
# Tests + Actions

An investigation on what tests can be used in CI.

- Jest "functional" tests:
    - Cheap.
    - Fast.
    - Probably best used in a unit context.
- Storybook interaction tests:
    - Based on Storybook play functions.
    - Some already exist in the project.
    - Very functional.
    - Easy to follow via the Storybook UI.
- Storybook visual tests:
    - Use Chromatic to identity slight visual changes between versions.
    - Can pick up on CSS related regressions that no other mechanism can.
    - Results can only be seen in the Chromatic dashboard, which is not public.
    - Probably not as important as other types of tests.
- Storybook integration (E2E) tests:
    - Use Cypress.
    - Nice DX with the Cypress UI.
    - Solid E2E option integrated in Storybook.
    - Would need to check how E2E they really are with a real use case.
- Existing Cypress E2E tests:
    - Already available, ready to use.
    - Are actually E2E, using a running Supabase instance.
